projects/components/src/load-async/wrapper/load-async-wrapper.component.ts

```diff
@@ -66,36 +79,53 @@ export class LoadAsyncWrapperComponent {
         this.loaderType = this.config?.load?.loaderType;
         break;
       case LoadAsyncStateType.NoData:
-        this.icon = this.showNoDataIcon() ? this.getNoDataIcon() : undefined;
-        this.title = this.config?.noData?.title ?? 'No Data';
-        this.description = this.config?.noData?.description ?? '';
+        this.updateNoDataMessage();
         break;
       case LoadAsyncStateType.GenericError:
-        this.icon = this.showErrorIcon() ? this.getErrorIcon() : undefined;
-        this.title = this.config?.error?.title ?? 'Error';
-        this.description = state.description ?? this.config?.error?.description ?? '';
+        this.updateGenericErrorMessage(state);
         break;
       default:
         this.icon = undefined;
         this.title = '';
         this.description = '';
+        this.customNoDataOrErrorTemplate = undefined;
+    }
+  }
+
+  private updateGenericErrorMessage(state: NoDataOrErrorStateConfig): void {
+    if (this.config?.error instanceof TemplateRef) {
+      this.customNoDataOrErrorTemplate = this.config?.error;
+    } else {
+      this.icon = this.showErrorIcon(this.config?.error) ? this.getErrorIcon(this.config?.error) : undefined;
+      this.title = this.config?.error?.title ?? 'Error';
+      this.description = state.description ?? this.config?.error?.description ?? '';
+    }
+  }
+
+  private updateNoDataMessage(): void {
+    if (this.config?.noData instanceof TemplateRef) {
+      this.customNoDataOrErrorTemplate = this.config?.noData;
+    } else {
+      this.icon = this.showNoDataIcon(this.config?.noData) ? this.getNoDataIcon(this.config?.noData) : undefined;
+      this.title = this.config?.noData?.title ?? 'No Data';
+      this.description = this.config?.noData?.description ?? '';
     }
   }
 
-  private getNoDataIcon(): IconType {
-    return this.config?.noData?.icon ?? IconType.NoData;
+  private getNoDataIcon(data?: NoDataOrErrorStateConfig): IconType {
+    return data?.icon ?? IconType.NoData;
   }
 
-  private getErrorIcon(): IconType {
-    return this.config?.error?.icon ?? IconType.Error;
+  private getErrorIcon(data?: NoDataOrErrorStateConfig): IconType {
```

Comment:
we should perhaps keep the current error and no data icons so that it is consistent when we want to show them.

else for template usecases, you'll have to supply the icons as well. its avoidable if we can default in this wrapper.

Reply:
would like to be careful here. We want to have consistency and that is why we did not expose the config early on. 
@adisreyaj could you please share the use case on slack?

Reply:
@anandtiwary Can you check the PR once again...made few more changes after the discussions.

Reply:
@anandtiwary ping on this ^^